### ISSUE TITLE
"Made Podfiles use symlinks to local pods" not working

### ISSUE THREAD

**Issue by burnssun** [flutter/flutter#15099]
## Steps to Reproduce
```shell
set -x

mkdir -p Made_Podfiles_use_symlinks_to_local_pods_NOT_WORKING/
pushd  Made_Podfiles_use_symlinks_to_local_pods_NOT_WORKING/

flutter create -t plugin -i swift --org com.failed failed > /dev/null
pushd failed/example
flutter build ios --debug 
popd

flutter create -t plugin -i swift --org com.success success > /dev/null
pushd success/example
wget -q https://raw.githubusercontent.com/mravn-google/flutter/3723ec0383f79420486bb4ee5acf06ebd190b5cf/packages/flutter_tools/templates/cocoapods/Podfile-swift
cp Podfile-swift ios/Podfile
flutter build ios --debug
popd

popd
```

## Flutter Doctor

➜  ~ flutter doctor -v
[✓] Flutter (Channel master, v0.1.8-pre.43, on Mac OS X 10.13.3 17D102, locale en-KR)
    • Flutter version 0.1.8-pre.43 at /Users/?????/flutter
    • Framework revision fe334e1652 (3 hours ago), 2018-03-02 17:54:51 -0800
    • Engine revision 97b22348c8
    • Dart version 2.0.0-dev.32.0.flutter-ee15c8eb68

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/?????/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = /Users/?????/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Community Edition (version 2017.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 22.1.2
    • Dart plugin version 173.4548.30

[✓] VS Code
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension version 2.8.0

[✓] Connected devices (1 available)
    • ????????????????????????????????????? • ios • iOS 11.2.6

• No issues found!

**Comment by xster** [comment]
Can you describe the failure mode? Does flutter build ios throw an error?

**Comment by anongit** [comment]
> Can you describe the failure mode? Does flutter build ios throw an error?

When using some plugins like barcode_scan it can't find the plugin header:

```objc
/volumes/data/code/example_app/ios/Runner/GeneratedPluginRegistrant.m:6:9: fatal error:
'barcode_scan/BarcodeScanPlugin.h' file not found
#import <barcode_scan/BarcodeScanPlugin.h>
```

Using the `3723ec03` version of Podspec file fixes the problem.

**Comment by mravn-google** [comment]
I can reproduce. This seems to be a problem with Swift projects. I cannot reproduce without the `-i swift` option.

**Comment by mravn-google** [comment]
There is a workaround described [here](https://stackoverflow.com/a/24317456/8059855)

**Comment by rmtmckenzie** [comment]
See #15031 for a more lengthy post about this. In particular the part about the [pull request](https://github.com/CocoaPods/CocoaPods/pull/7470) I made to cocoapods to fix their issue and the [relevant issue](https://github.com/CocoaPods/CocoaPods/issues/7463). The issue should be reproducible for any project using `use_frameworks!`.

**Comment by willks** [comment]
I too have experienced this issue and verified that it's only with Swift. I must state that we could not replicate issue with ObjC. I'm trying to write wrappers for existing frameworks in order for me to justify going full Flutter for all future development. I'm still committed, but not putting it forward until I can make plugins for existing Swift projects (think Couchbase-Lite, which is my first attempt). We don't want to touch ObjC.

**Comment by mravn-google** [comment]
To apply the fix in #15437 to a Swift-based Flutter project, add [these lines](https://github.com/flutter/flutter/pull/15437/files#diff-519f1341a7393048261b00690f7c2c0e) to `ios/Podfile`.

**Comment by burnssun** [comment]
@mravn-google 👍 #15437 WORKS!!!!
